components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5541,23 +5541,29 @@ private Attribute getSchemaAttributes(ExternalClaim scimClaim, LocalClaim mapped
                                           boolean isExtensionAttr) {
 
         String name = scimClaim.getClaimURI();
+        String claimDielectURI = scimClaim.getClaimDialectURI();
+        boolean isCustomScemaAttr = claimDielectURI.equalsIgnoreCase(getCustomSchemaURI());
         if (name.startsWith(scimClaim.getClaimDialectURI())) {
             name = name.substring(scimClaim.getClaimDialectURI().length() + 1);
         }
 
         AbstractAttribute attribute;
-        if (isComplexAttribute(name)) {
+        if (isComplexAttribute(name, isCustomScemaAttr)) {
             attribute = new ComplexAttribute(name);
         } else {
             attribute = new SimpleAttribute(name, null);
         }
 
-        populateBasicAttributes(mappedLocalClaim, attribute, isExtensionAttr);
+        populateBasicAttributes(mappedLocalClaim, attribute, isExtensionAttr, isCustomScemaAttr);
 
         return attribute;
     }
 
-    private boolean isComplexAttribute(String name) {
+    private boolean isComplexAttribute(String name, boolean isCustomScemaAttr) {
```

Comment:
This logic can be decided even before calling this method. Isn't it ? There is no inhouse login for this decision. WDYT ?

Reply:
Fix with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/459/commits/b5eaab3e1944238b0c0282a38f332f22509a18f5